### ISSUE TITLE
Drones using build schematics 

### ISSUE THREAD

**Issue by desht** [TeamPneumatic/pnc-repressurized#91]
Copied from the Curseforge page:

Thank you very much for making this wonderful mod available for 1.12.2!
I don't know if you even plan to add new stuff to the mod or if you are allready tired of people who want you to put extra stuff in it. So just ignore this in that case ;).

My idea:

If drones could "scan" areas and if you feed the drones with the right items, they will rebuild the areas in a different location. Even better would it be, if you could save those areas (like you already can with puzzle programms) so your drones could rebuild them on new maps or you could share those saves with other people.

Just imagine to share base parts, whole bases or multiblock setups with friends and your drones would build them for you.

I have no clue how much work this is or if this is possible at all but i think it would make the mod to a must have mod in most modpacks.

Anyway, thank you for the great work and sorry for my bad english.

**Comment by Tekstack** [comment]
A couple years ago, (in 1.7.10) I built a relative coordinates drone program using the features in-game and shared to Pastebin. Both of these features are working properly in the 1.12 build and can be used to make programs as mentioned above from a single gps cord, but you must build the program first (I did in sections and tested as I went), then apply the relative cords...this may sound like a bit of a ramble and a bit hard to follow, but the main point is, we already can build and share builds that can be placed anywhere in the world(s).  However, there isn't a good place to share these builds with others, and initial setup to create the build to share takes a lot of time to put together vs simply building something by hand then scanning the area. Below is a video that I took showing this concept off. It isn't perfect, but it uses the Aphorism Tile as a guide to the user to input certain items such as glass, stone etc. There is a pastebin link in the description that  doesn't work in the 1.12 version due to so many changes, but it should work still on 1.7.10 if you want to see it in game.
https://www.youtube.com/watch?v=sGnLgVPfxmE

**Comment by MineMaarten** [comment]
That's awesome @Tekstack ! This does work well if you don't have too many different block types, so you can express the different areas needing to be filled with a 'manageable' amount of puzzle pieces. I do think this proposed feature would make it a lot easier, to encode an area into a puzzle piece (blueprint), and to let the drone redo it elsewhere.

We shouldn't underestimate how hard it is though, this is basically BuildCraft's Blueprint system, and they have had many problems with it in the past, most notably rotation logic and preventing dupe bugs (figuring out which block requires which item to place it).

It would be very cool nevertheless :)

**Comment by Asherslab** [comment]
hey, just wanted to put it out there, we do something similar in concept to this with our own mod's entities over here: 
Builder: https://github.com/Minecolonies/minecolonies/tree/version/1.12/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder
Scanner: https://github.com/Minecolonies/minecolonies/blob/version/1.12/src/main/java/com/minecolonies/coremod/items/ItemScanTool.java

Feel free to check that out for any inspiration =D, some of that code probably could be reworked as of recently. but i definitely know that i love to have a jumping off point for starting an ambitious project
